sqlitecluster/SQLiteNode.cpp

```diff
@@ -74,6 +80,143 @@ SQLiteNode::~SQLiteNode() {
     SASSERTWARN(!commitInProgress());
 }
 
+void SQLiteNode::replicate(SQLiteNode& node, SQLite& db, int threadNum) {
```

Comment:
I'd love a comprehensive comment here explaining at a high level how all this works.  Something like:

> Whenever a node is FOLLOWING, the sync thread manages one or more "replication threads".  These wait for activity on the `_replicationCommands` queue, which notifies the `_replicationCV` condition variable every time it's changed.  Then it takes the appropriate action on each replication command, which can be one of the following:
> 
> - BEGIN_TRANSACTION - Waits for the global database commit to be one less than this new command's commit count (ie, for when it's this transaction's turn to be run), then starts this transaction.  The thread then waits until this command's hash appears in either the `_rollbackHashes` or `_replicationHashes` datastructures, which indicate this thread should either rollback or commit the transaction.
> 
> - ROLLBACK_TRANSACTION - Inserts the replication hash into `_rollbackHashes`, instructing the thread that has begun this transaction to roll it back.
> 
> - COMMIT_TRANSACTION - Inserts the replication hash into `_replicationHashes`, instructing the thread that has begun this transaction to commit it.